kbfscrypto/encrypted_data.go

```diff
@@ -183,12 +214,50 @@ func EncryptPaddedEncodedBlock(paddedEncodedBlock []byte, key BlockCryptKey) (
 	return EncryptedBlock{encryptedData}, nil
 }
 
+// EncryptPaddedEncodedBlockV2 encrypts a padded, encoded block, using
+// the v2 block encryption scheme.
+func EncryptPaddedEncodedBlockV2(paddedEncodedBlock []byte, key BlockHashKey) (
+	encryptedBlock EncryptedBlock, err error) {
+	encryptedData, err := encryptDataWithNonce(
+		paddedEncodedBlock, key.cryptKey(), key.nonce(),
+		EncryptionSecretboxWithKeyNonce)
+	if err != nil {
+		return EncryptedBlock{}, err
+	}
+
+	return EncryptedBlock{encryptedData}, nil
+}
+
 // DecryptBlock decrypts a block, but does not unpad or decode it.
 func DecryptBlock(encryptedBlock EncryptedBlock, key BlockCryptKey) (
 	[]byte, error) {
+	if encryptedBlock.encryptedData.Version == EncryptionSecretboxWithKeyNonce {
+		return nil, errors.WithStack(
+			InvalidEncryptionVer{encryptedBlock.encryptedData.Version})
+	}
+
 	return decryptData(encryptedBlock.encryptedData, key.Data())
 }
 
+// DecryptBlockV2 decrypts a block, but does not unpad or decode it.
+func DecryptBlockV2(encryptedBlock EncryptedBlock, key BlockHashKey) (
```

Comment:
is this only called in test?

Reply:
I start calling it for real in the next PR, #1837.